analyzers/src/SonarAnalyzer.CSharp/Rules/MemberInitializerRedundant.cs

```diff
@@ -82,24 +85,17 @@ private void CheckInstanceMembers(SonarSyntaxNodeReportingContext c, TypeDeclara
                 return;
             }
 
-            var initializerDeclarations = GetInitializerDeclarations<ConstructorDeclarationSyntax>(c, typeInitializers);
```

Comment:
Everything that follows from this point should be considered a re-factoring and reviewed as such.